modules/ggl-uri/src/gghttp_uri.c

```diff
@@ -136,3 +138,283 @@ GglError gg_uri_parse(GglArena *arena, GglBuffer uri, GglUriInfo *info) {
 
     return GGL_ERR_OK;
 }
+
+static GglError find_docker_uri_separators(
+    GglBuffer uri,
+    size_t slashes[2],
+    size_t *slash_count,
+    size_t colons[3],
+    size_t *colon_count,
+    size_t *at,
+    bool *has_registry
+) {
+    for (size_t i = uri.len; i > 0; i--) {
```

Comment:
A minor suggestion: make this like so
```suggestion
    for (size_t len = uri.len - 1; len >= 0; len--) {
```

This will allow you to use `len` everywhere instead of `len - 1`.

Reply:
You'd need to check before though that `uri.len` is non 0. And return apt error code if that is the case.

Reply:
This was written this way since size_t is always positive so there's a warning if I write the for loop the other way as `len >= 0` is always true.

Reply:
Yes, I missed that. Thanks for clarifying.